ecommerce/serializers.py

```diff
@@ -694,3 +729,87 @@ def get_consent_text(self, instance):
     class Meta:
         fields = ["consent_date", "id", "company", "consent_text"]
         model = models.DataConsentUser
+
+
+class LineSummarySerializer(serializers.ModelSerializer):
+    """ Summary serializer for Line model """
+
+    product_version = ProductVersionSummarySerializer()
+
+    class Meta:
+        model = models.Line
+        fields = ["product_version", "quantity"]
+
+
+class OrderReceiptSerializer(serializers.ModelSerializer):
+    """ Serializer for extracting receipt info from an Order object"""
+
+    lines = serializers.SerializerMethodField()
+    purchaser = serializers.SerializerMethodField()
+    coupon = serializers.SerializerMethodField()
+    order = serializers.SerializerMethodField()
+    receipt = serializers.SerializerMethodField()
+
+    def get_receipt(self, instance):
+        """Get receipt information if it exists"""
+        receipt = instance.receipt_set.order_by("-created_on").first()
+        if receipt:
+            data = dict(card_number=receipt.data["req_card_number"], card_type=None)
```

Comment:
what would be the value of it if in cased of 100% free coupon? Will data have that card detail in the receipt?

Reply:
Receipt is created for CyberSource orders. This in turn menas if it's a $0.0 order it doesn't go to CyberSource and there is no payment/receipt. So, if there's a receipt for an order, there's card information available. Although I can add those checks if you suggest.

Reply:
if card information will be there for sure then it's not needed.